### PR TITLE
ci: setup GitHub Actions

### PR DESCRIPTION
**Pull request by nschonni**
Was hitting some macOS issues that I thought might be coming from here, so i thought it might be good to add back a full test matrix.
I left in the Azure DevOps CI, but only added in 12+ node versions since Node 10 is EOL

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      - name: 📥 Download deps
+        run: npm ci
+
+      - name: Run coverage
+        run: npm test
+
+      - name: Run browser tests
+        run: npm run mocha-browser-test -- --browsers ${{ matrix.browser }}  --timeout 10000
+        env:
+          DISPLAY: :99.0
```
Comment (nschonni) [inline .github/workflows/ci.yml:64]: This might not be needed anymore since switching to FirefoxHeadless

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+      fail-fast: false
+      matrix:
+        node:
+          - 12
+          - 14
+          - 16
+          - 17
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        browser:
+          - FirefoxHeadless
```
Comment (nschonni) [inline .github/workflows/ci.yml:38]: Edge and Safari runners seems to be abandoned and a little flaky, so didn't add them in, but can be added on platform specific `includes:`

**Review by nschonni — COMMENTED** [review]
Can see the run from my fork https://github.com/nschonni/async/actions/runs/1668517012

**Comment by aearly** [comment]
Thanks for doing this, I didn't realize Travis has basically shut down.

**Pull request merged**